.github/workflows/main.yml

```
@@ -0,0 +1,29 @@
+name: Build & Test
+
+on:
+  push:
+    branches: [trunk]
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [18.x, 20.x]
+      fail-fast: false
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Use Node v${{ matrix.node-version }}
+        uses: actions/setup-node@v4
+        with:
+          cache: 'pnpm'
+          node-version: ${{ matrix.node-version }}
+      - name: Use pnpm
+        uses: pnpm/action-setup@v4
+        run_install: false
+      - name: Install
+        run: pnpm i
+      - name: Build
+        run: pnpm run -r build
```

```suggestion
        run: pnpm run --recursive build
```

Ditto with my other review to replace `pnpm i`